packages/design/src/DataTable/Table.test.tsx

```diff
@@ -319,3 +319,49 @@ describe('sorting by field defined in key and altSortKey', () => {
     expect(cells[2]).toHaveTextContent('1');
   });
 });
+
+
+describe("sorting by iso date string", () => {
+  const sample = [
+    {
+      created: new Date(12004).toISOString(),
+      text: 'a2',
+    },
+    {
+      created: new Date(12002).toISOString(),
+      text: 'a3',
+    },
+    {
+      created: new Date(101012004).toISOString(),
+      text: 'a1',
+    },
+  ];
+  
+  test("sort by iso date string", () => {
+    render(
+      <Table
+        data={sample}
+        columns={[
+          {
+            key: 'created',
+            headerText: 'created',
+            isSortable: true,
+          },
+          {
+            key: 'text',
+            headerText: 'text',
+            isSortable: false,
+          },
+        ]}
+        emptyText=""
+        initialSort={{ key: 'created', dir: 'DESC' }}
+      />
+    );
+
+    const { rows } = getTableRows();
+
+    expect(rows[0]).toHaveTextContent('a1');
+    expect(rows[1]).toHaveTextContent('a2');
+    expect(rows[2]).toHaveTextContent('a3');
+  });
+});
```

Comment:
Nit
line 339: ESLint: Trailing spaces not allowed.(no-trailing-spaces)
EOF: no new line